content/learn/book/migration-guides/0.8-0.9/_index.md

```diff
@@ -0,0 +1,845 @@
+
+```rust
+// 0.8
+fn scoped_function<'scope>(scope: &mut Scope<'scope, ()>) {}
+
+// 0.9
+fn scoped_function<'scope>(scope: &Scope<'_, 'scope, ()>) {}
+```
+
+`scope.spawn_local` changed to `scope.spawn_on_scope` this should cover cases where you needed to run tasks on the local thread, but does not cover spawning Nonsend Futures. Spawning of NonSend futures on scope is no longer supported.
+
+### [Extract Resources into their own dedicated storage](https://github.com/bevyengine/bevy/pull/4809)
+
+Resources have been moved to `Resources` under `Storages` in `World`. All code dependent on `Archetype::unique_components(_mut)` should access it via `world.storages().resources()` instead.
+
+All APIs accessing the raw data of individual resources (mutable _and_ read-only) have been removed as these APIs allowed for unsound unsafe code. All usages of these APIs should be changed to use `World::{get, insert, remove}_resource`.
+
+### [Clean up Fetch code](https://github.com/bevyengine/bevy/pull/4800)
+
+Changed: `Fetch::table_fetch` and `Fetch::archetype_fetch` have been merged into a single `Fetch::fetch` function.
+
+### [Rename `ElementState` to `ButtonState`](https://github.com/bevyengine/bevy/pull/4314)
+
+The `ElementState` type received a rename and is now called `ButtonState`. To migrate you just have to change every occurrence of `ElementState` to `ButtonState`.
+
+### [Fix incorrect and unnecessary normal-mapping code](https://github.com/bevyengine/bevy/pull/5766)
+
+`prepare_normal` from the `bevy_pbr::pbr_functions` shader import has been reworked.
+
+Before:
+
+```wgsl
+    pbr_input.world_normal = in.world_normal;
+
+    pbr_input.N = prepare_normal(
+        pbr_input.material.flags,
+        in.world_normal,
+#ifdef VERTEX_TANGENTS
+#ifdef STANDARDMATERIAL_NORMAL_MAP
+        in.world_tangent,
+#endif
+#endif
+        in.uv,
+        in.is_front,
+    );
+```
+
+After:
+
+```wgsl
+    pbr_input.world_normal = prepare_world_normal(
+        in.world_normal,
+        (material.flags & STANDARD_MATERIAL_FLAGS_DOUBLE_SIDED_BIT) != 0u,
+        in.is_front,
+    );
+
+    pbr_input.N = apply_normal_mapping(
+        pbr_input.material.flags,
+        pbr_input.world_normal,
+#ifdef VERTEX_TANGENTS
+#ifdef STANDARDMATERIAL_NORMAL_MAP
+        in.world_tangent,
+#endif
+#endif
+        in.uv,
+    );
+```
+
+### [Replace root list with struct](https://github.com/bevyengine/bevy/pull/6354)
+
+The scene file format now uses a struct as the root object rather than a list of entities. The list of entities is now found in the `entities` field of this struct.
+
+```rust
+// Old (Bevy 0.8)
+[
+  (
+    entity: 0,
+    components: [
+      // Components...
+    ]
+  ),
+]
+
+// New (Bevy 0.9)
+(
+  entities: [
+    (
+      entity: 0,
+      components: [
+        // Components...
+      ]
+    ),
+  ]
+)
+```
+
+### [Use map for scene `components`](https://github.com/bevyengine/bevy/pull/6345)
+

```

Comment:
All the scene format changes should probably be combined into 1 entry.

Reply:
Can you make a PR to my branch with those changes? Otherwise I'll do it tomorrow.

Reply:
I probably won't get to it today. If you don't get to it tomorrow, I can make a PR.

Reply:
If anyone reads this and wants to do it, just open a PR to my fork with the changes and I'll merge it.